src/frameworks/compose/index.ts

```diff
@@ -1,31 +1,36 @@
 import { AppBundle } from "./interfaces";
 import { getDriver, Mode } from "./driver";
 import { discover } from "./discover";
+import { FrameworkSpec, FileSystem } from "./discover/types";
 
 /**
  * Run composer in the specified execution context.
  */
-export function compose(mode: Mode): AppBundle {
+export async function compose(
+  mode: Mode,
+  fs: FileSystem,
+  allFrameworkSpecs: FrameworkSpec[]
+): Promise<AppBundle> {
   let bundle: AppBundle = { version: "v1alpha" };
-  const spec = discover();
+  const spec = await discover(fs, allFrameworkSpecs);
   const driver = getDriver(mode, spec);
 
-  if (spec.startCommand) {
+  if (spec.detectedCommands && spec.detectedCommands.run) {
```

Comment:
nit* same as `if (spec.detectedCommands?.run)`

(does TS not like this?)

Reply:
Previously, `spec` can be `null` if `package.json` is not present. So TS doesn't like `?` symbol on optionals and suggested to use `spec && spec.detectedCommands && spec.detectedCommands.run`. 

Later, I changed so that `spec` won't be null even if `package.json` is not found. It contains other useful parameter like `install` command etc. So, we can use `?` notation.